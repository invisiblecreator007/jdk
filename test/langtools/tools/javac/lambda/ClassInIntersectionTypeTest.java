/*
 * @test /nodynamiccopyright/
 * @bug 8322810
 * @summary Lambda expressions can implement classes
 * @compile/fail/ref=ClassInIntersectionTypeTest.out -XDrawDiagnostics ClassInIntersectionTypeTest.java
 */

import java.io.Serializable;

public class ClassInIntersectionTypeTest {
    // test 1
    void m() {
        var r = (ClassInIntersectionTypeTest & Runnable) () -> System.out.println("Hello, World!");
    }

    // test 2
    static void foo() {
        run(() -> System.out.println("Hello, World!"));
    }

    static <T extends ClassInIntersectionTypeTest & Runnable> void run(T t) {
        t.run();
    }

    @interface Anno {}
    Anno a = (Anno & Serializable) ()-> null; // OK

    Anno b = ()-> null; // OK

    static void bar() {
        annotationType(() -> null);
    }
    static <T extends Anno & Serializable> void annotationType(T t) {
        t.annotationType();  // OK
    }
}
