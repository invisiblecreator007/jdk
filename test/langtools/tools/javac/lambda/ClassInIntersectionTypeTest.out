ClassInIntersectionTypeTest.java:11:58: compiler.err.prob.found.req: (compiler.misc.bad.intersection.target.for.functional.expr: (compiler.misc.not.an.intf.component: ClassInIntersectionTypeTest))
ClassInIntersectionTypeTest.java:16:12: compiler.err.prob.found.req: (compiler.misc.infer.no.conforming.assignment.exists: T, (compiler.misc.bad.intersection.target.for.functional.expr: (compiler.misc.not.an.intf.component: ClassInIntersectionTypeTest)))
2 errors
